app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -107,6 +107,10 @@ def machine_id(feed)
       os_resource_for(feed).try(:properties).try { |prop| prop['Machine Id'] }
     end
 
+    def container_id(feed)
```

Comment:
I'd extract all resource-related stuff into it's own class, but I'd also keep it on providers/hawkular. There's no need to put it further away from where it is actually being used.